src/disnake/ext/plugins/plugin.py

```diff
@@ -753,6 +749,32 @@ async def _before_loop():
 
         return decorator
 
+    def loop(
+        self, *, wait_until_ready: bool = True, **loop_args: t.Any
```

Comment:
Not really a fan of the **kwargs here... Ideally we could copy the existing typing on the disnake loop decorator. The main reason I didn't originally merge these two decorators is that this way the typing of the loop decorator is entirely on disnake's end, and if it ever does update we don't get needless version incompatibilities.

Reply:
If we're gonna have a 100% correct and up-to-date typings for @tasks.loop kwargs, then i agree this pr is actually a bad idea. In our case we didn't really care about full typing, and as such the implementation was very simple. But to support full typing we'd need to import whole lot of additional types and as you correctly pointed out this will require quite a big effort to keep up with. I don't think this is worth it for a feature that is usually used two-three times at most through entire codebase of an average bot. For now i copied the docstring and argument typings from the default impl of tasks.Loop, but not overloads. This disallows usage of Loop subclasses that have extra arguments besides the default ones.